src/control/system/mdb.go

```diff
@@ -0,0 +1,151 @@
+//
+// (C) Copyright 2020 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package system
+
+import (
+	"encoding/json"
+	"net"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+)
+
+type (
+	// MemberRankMap provides a map of Rank->*Member.
+	MemberRankMap map[Rank]*Member
+	// MemberUuidMap provides a map of UUID->*Member.
+	MemberUuidMap map[uuid.UUID]*Member
+	// MemberAddrMap provides a map of net.Addr->[]*Member.
+	MemberAddrMap map[net.Addr][]*Member
+
+	// MemberDatabase contains a set of maps for looking
+	// up members and provides methods for managing the
+	// membership.
+	MemberDatabase struct {
+		Ranks MemberRankMap
+		Uuids MemberUuidMap
+		Addrs MemberAddrMap
+	}
+)
+
+// MarshalJSON creates a serialized representation of the MemberRankMap.
+// The member's UUID is used to represent the member in order to
+// avoid duplicating member details in the serialized format.
+func (srm MemberRankMap) MarshalJSON() ([]byte, error) {
```

Comment:
to align with the convention we have been using this should be "mrm" as the receiver

Reply:
Whoops. Yeah, in a previous version of this I was calling these Server*Map, forgot to update the receivers.